components/builder-web/app/util.ts

```diff
@@ -32,6 +32,32 @@ export function createGitHubLoginUrl(state) {
   return `${urlPrefix}?${queryString}`;
 }
 
+export function createLoginUrl() {
+  const store = new AppStore();
+  let params: any = {};
+  let uri;
+
+  switch (config.oauth_provider) {
+    case 'github':
```

Comment:
Is there a default case we can add for generic oauth providers?

Reply:
This sounds good, but I fear would be pretty difficult in practice.  Bitbucket's setup and URLs are different from GitHub's and I bet if we checked Gitlab, those would be slightly different as well.  I'm not sure there's a "standard" set of URLs for oauth.

Reply:
That's right, there isn't really a default case from our perspective -- there are conventions for OAuth paths and param names and the like, but that's about it. Our list of supported providers will also be explicit, and as of now, it's still only GitHub.